packages/lesswrong/server/rsvpToEvent.ts

```diff
@@ -10,13 +10,27 @@ const responseSortOrder = {
   no: 3
 }
 
+type ValidResponse = 'yes' | 'no' | 'maybe';
+const isValidResponse = (response: string): response is ValidResponse => {
+  return response in responseSortOrder;
+}
+
 addGraphQLMutation('RSVPToEvent(postId: String, name: String, email: String, private: Boolean, response: String): Post');
 addGraphQLResolvers({
   Mutation: {
     async RSVPToEvent(root: void, {postId, name, email, nonPublic, response}: {postId: string, name: string, email: string, nonPublic: boolean, response: string}, context: ResolverContext) {
       const { currentUser } = context;
+      
+      if (!currentUser) {
+        throw new Error('Error submitting RSVP: Not logged in');
+      }
+
+      if (!isValidResponse(response)) {
```

Comment:
This we do actually need since we didn't have any sort of runtime validation (except, maybe, simpl-schema itself?) that a valid response was being provided.  GraphQL doesn't allow for union types of string literals, so we can't push it up to that layer.